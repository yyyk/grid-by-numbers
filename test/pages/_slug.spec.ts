import Vue from 'vue'
import Vuetify from 'vuetify'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import SlugPage from '@/pages/_slug.vue'

Vue.use(Vuetify)

const localVue = createLocalVue()

describe('_slug.vue', () => {
  let vuetify: any

  const mountFunction = (options = {}) => {
    return shallowMount(SlugPage, {
      localVue,
      vuetify,
      ...options
    })
  }

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('renders correctly', () => {
    const wrapper = mountFunction({
      data: () => ({
        doc: null,
        prev: null,
        next: null
      })
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('exists', () => {
    const wrapper = mountFunction({
      data: () => ({
        doc: null,
        prev: null,
        next: null
      })
    })
    expect(wrapper.exists()).toBe(true)
  })

  it('should have nuxt-content when doc is not null', () => {
    const wrapper = mountFunction({
      data: () => ({
        doc: {},
        prev: null,
        next: null
      }),
      stubs: ['nuxt-content']
    })
    expect(wrapper.exists()).toBe(true)
  })
})
