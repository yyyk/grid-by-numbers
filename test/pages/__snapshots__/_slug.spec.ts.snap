// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_slug.vue renders correctly 1`] = `
"<v-container-stub tag=\\"div\\">
  <v-layout-stub tag=\\"div\\" column=\\"\\" justify-start=\\"\\" align-stretch=\\"\\">
    <v-flex-stub tag=\\"div\\" xs12=\\"\\" sm12=\\"\\" md12=\\"\\">
      <!---->
      <pager-stub></pager-stub>
    </v-flex-stub>
  </v-layout-stub>
</v-container-stub>"
`;
